File: TaskManager/Lab_final_210042166/Task_Management_System/Add_Comment/AddCommentCommand.java
package Lab_final_210042166.Task_Management_System.Add_Comment;

import Lab_final_210042166.Task_Management_System.Comment.Comment;
import Lab_final_210042166.Task_Management_System.Interface.Command;
import Lab_final_210042166.Task_Management_System.Issue.Issue;

public class AddCommentCommand  implements Command {
  private Issue issue;
  private Comment comment;

  public AddCommentCommand(Issue issue,Comment comment)
  {
      this.issue=issue;
      this.comment=comment;
  }


    @Override
    public void execute()
    {
        issue.addComment(comment);
    }

    @Override
    public void undo()
    {
        issue.getComments().remove(comment);
    }
}

File: taskManager/labfinal_210042111/AddCommentCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace labfinal_210042111
{
    public class AddCommentCommand: ICommand
    {
        private IssueRepository repository;
        private Issue issue;
        private Comment comment;
        public AddCommentCommand(IssueRepository repository, Issue issue, Comment comment)
        {
            this.repository = repository;
            this.issue = issue;
            this.comment = comment;
        }
       
        public void Execute()
        {
            issue.AddComments(comment);
        }
     
    public string undo()
        {
            return "sucess";
            // Code to remove the last comment
        }
    }
}
